File: problems/344/reverse_string.go
package reverse

// ReverseString reverses an array without using any auxilary space
func ReverseString(input []string) (reversed []string) {
	if len(input) <= 1 {
		return input
	}

	inputLength := len(input) - 1
	timesRun := inputLength / 2
	for i := 0; i <= timesRun; i++ {
		input[i], input[inputLength-i] = input[inputLength-i], input[i]
	}
	return input
}

File: problems/344/reverse_string.py
# Reverse using two pointers technique.


def reverse_string(string):
    return list(reversed(string))


def reverse_string_two_pointers(string):
    pointer_one = 0
    pointer_two = len(string) - 1

    while pointer_one < pointer_two:
        string[pointer_one], string[pointer_two] = (
            string[pointer_two],
            string[pointer_one],
        )

        pointer_one += 1
        pointer_two -= 1

    return string
